Bugfix to MixtureSameFamily's _pad_mixture_dimension

CJMenart opened the pull request:
Fixes Issue #73792

This is a duplicate of pull request. #73864. It's a small bugfix that should have happened a long time ago, but it didn't because I didn't actually follow up with the pull request after originally submitting. That's my bad. Trying to remedy the error.

This contains a fix to _pad_mixture_dimension, which intends to count the number of dimensions in its referent tensors, but accidentally counts the number of elements (and can thus end up creating tensors with potentially thousands of dimensions by mistake). Also contains a single test for the fixed behavior.

cc @albanD

CJMenart committed:
Bugfix to _pad_mixture_dimensions

Bugfix to _pad_mixture_dimensions. Now counts number of dimensions instead of number of elements as intended.

CJMenart committed:
+test for _pad_dimensions bugfix

Regression test for bugfix to _mad_mixture_dimensions in previous commit.

soulitzer commented on test/distributions/test_distributions.py:
```diff
@@ -4048,6 +4048,11 @@ def test_continuous_bernoulli_shape_tensor_params(self):
+        mix_distribution = Categorical(torch.ones([3,1,3]))
+        component_distribution = torch.distributions.Normal(torch.zeros([3,3,3]), torch.ones([3,3,3]))
+        gmm = MixtureSameFamilyFixed(mix, comp)
+        self.assertEqual( len(gmm.mean.shape), 2)
```
```suggestion
        self.assertEqual(len(gmm.mean.shape), 2)
```

soulitzer commented on test/distributions/test_distributions.py:
```diff
@@ -4048,6 +4048,11 @@ def test_continuous_bernoulli_shape_tensor_params(self):
+    def test_mixture_same_family_shape(self):
+        mix_distribution = Categorical(torch.ones([3,1,3]))
+        component_distribution = torch.distributions.Normal(torch.zeros([3,3,3]), torch.ones([3,3,3]))
+        gmm = MixtureSameFamilyFixed(mix, comp)
```
whoops looks like you forgot to rename these variables

CJMenart replied:
...My God. This is what I get for not testing the exact same text I commit...chuckleheads like me are why we _need_ devops.

CJMenart committed:
Fix var names in test_mixture_same_family_mean_shape

CJMenart committed:
MixtureSameFamilyFixed ->MixtureSameFamily

CJMenart committed:
remove non-PIP space

Co-authored-by: Jeffrey Wan <soulitzer@gmail.com>

soulitzer reviewed (approved):
Thanks!

soulitzer commented:
@pytorchbot merge

CJMenart committed:
spaces after commas

As caught by CI

soulitzer commented:
@pytorchbot merge

CJMenart commented:
So two checks have failed on the new test. Both are ones with Torch Dynamo. I don't really know anything about Dynamo, or why it's accessing non-existent Tensor attributes. A lot of tests for torch.distributions are marked with this "Don't test this with Dynamo" tag. Trying to figure out if it's appropriate to apply that to the new one.

soulitzer commented:
> So two checks have failed on the new test. Both are ones with Torch Dynamo. I don't really know anything about Dynamo, or why it's accessing non-existent Tensor attributes. A lot of tests for torch.distributions are marked with this "Don't test this with Dynamo" tag. Trying to figure out if it's appropriate to apply that to the new one.

Adding the skip like the other tests do sounds good to me

CJMenart committed:
Skip test_mixture_same_family_mean_shape in Dynamo

Not knowledgeable about dynamo really; making a call because test seems simple/reasonable and many other torch.distribution tests skip if Dynamo.

CJMenart commented:
Tests are passed, do you think this is ready to go?

soulitzer commented:
Yup, thanks for the follow up

soulitzer commented:
@pytorchbot merge
